pkg/cloud/metadata/k8s.go

```diff
@@ -114,17 +116,70 @@ func KubernetesAPIInstanceInfo(clientset kubernetes.Interface) (*Metadata, error
 		return nil, fmt.Errorf("error getting Node %v: %w", nodeName, err)
 	}
 
-	providerID := node.Spec.ProviderID
-	if providerID == "" {
-		return nil, errors.New("node providerID empty, cannot parse")
+	numAttachedENIs := 1        // Default: All nodes have at least 1 attached ENI
+	numBlockDeviceMappings := 0 // Default: 0
+	sageMakerLabels := true
```

Comment:
I think the code would be easier to reason about and more correct if this defaulted to `false`